src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,624 @@
+
+Permanova assumes there is homogeneity of dispersion among groups. We want to also test this assumption, to check whether the differences detected by our permanova test are due to variation in dispersion of the data. We do this using the betadisper method to calculate variances and then evaluate if there’s a significant variation between groups through an anova.
+
+To do all these tests, we use the `addRDA` method, which will do all of them for us.
+
+```{r}
+beta_diversity_tse <- addRDA(
+  filtered_tse,
+  formula = data ~ sample_geographic.location..country.and.or.sea.region.,
+  assay.type = "relabundance",
+  method = "bray",
+  test.signif = TRUE,
+  homogeneity.test = "anova",
+  full = TRUE
+)
+```
+
+We now plot and fetch the results
+
+```{r}
+# Plot the results
+plotReducedDim(beta_diversity_tse, "RDA", colour_by = "sample_geographic.location..country.and.or.sea.region.")
+
+# Fetch significance results, including PERMANOVA and betadisper (homogeneity) results
+significance_results <- attr(reducedDim(beta_diversity_tse, "RDA"), "significance")
+significance_results
+```
+
+ANOVA's p-value is significant (p>0.05), meaning that there's no significant difference in dispersion among groups. The PERMANOVA however showed a significant P value (p<0.05), meaning significant differences in community composition between the groups. 
+
+## Detection of differentially abundant taxa
+There are many approaches out there to detect differentially abundant taxa (or genes). Here we detect microbiome community composition changes due to various environmental factors. Describing such associations in quantitative models has multiple applications, such as in predicting disease status based on microbiome data. In addition, the pipeline includes the generation of high-quality plots for visual inspection of results ([9](#ref-wirbel2021)). There are multiple parameters that can be adjusted for better performance in each step of the analysis. However, in this exercise we are using default arguments with minimal tuning.
+
+### Association testing
+
+1. First we use our original counts data on our `filtered_tse` object and agglomerate by rank to a new `genus_tse` object.
+
+```{r}
+genus_tse <- agglomerateByRank(filtered_tse, assay.type = "counts", rank = "Genus")
+```
+
+2. Now take the top 25% of taxa.
+
+```{r}
+genus_tse <- subsetByPrevalent(genus_tse, prevalence = 0.25, include.lowest = TRUE)
+```
+
+3. Now we transform the count assay to relative abundances with a relative abundance transformation, then do abundance filtering.
+
+```{r}
+# Apply relabundance transformation
+genus_tse <- transformAssay(genus_tse, assay.type = "counts", method = "relabundance")
+
+# Apply abundance filtering
```

Comment:
Prevalence rather than abundance filtering?

It seems confusing to have subsetByPrevalent on row 443 and then again here on row 453. Could we have just one?